pathway-analysis/01-overrepresentation_analysis.Rmd

```diff
@@ -107,60 +116,89 @@ In this example, we will use canonical pathways which are ([ref](https://www.gse
 And are a subset of `C2: curated gene sets`.
 Specifically, we will use the [KEGG (Kyoto Encyclopedia of Genes and Genomes)](https://www.genome.jp/kegg/) pathways. 
 
-```{r}
-# Filter the human data frame to the KEGG pathways that are included in the
+```{r filter_to_kegg}
+# Filter the mouse data frame to the KEGG pathways that are included in the
 # curated gene sets
-hs_kegg_df <- hs_msigdb_df %>%
+mm_kegg_df <- mm_msigdb_df %>%
   dplyr::filter(gs_cat == "C2",  # curated gene sets 
                 gs_subcat == "CP:KEGG")  # KEGG pathways 
 ```
 
 *Note: We could specified that we wanted the KEGG gene sets using the `category` and `subcategory` arguments of `msigdbr()`, but we're going for general steps!*
 
-```{r hs_kegg_columns}
-colnames(hs_kegg_df)
+```{r mm_kegg_columns}
+colnames(mm_kegg_df)
 ```
 
 The `clusterProfiler` function we will use requires a data frame with two columns, where one column contains the term identifier or name and one column contains gene identifiers that match our gene lists we want to check for enrichment.
 Our data frame with KEGG terms contains Entrez IDs and gene symbols.
 
-## Read in marker data and prep
+## Read in DGE results and prep
+
+```{r read_in_dge_results, live = TRUE}
+vs_low_df <- readr::read_tsv(vs_low_file)
+vs_unsorted_df <- readr::read_tsv(vs_unsorted_file)
+```
+
+Let's take a peek at the top of one of the results data frames.
+
+```{r peek_at_dge_results, live = TRUE}
+head(vs_low_df)
+```
+
+Some of these rows have `NA` values in `padj`, which [can happen for a number of reasons](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA) including when all samples have zero counts.
+
+Let's filter to rows that _do not have any `NA`_ using a function called `complete.cases()`.
+
+```{r complete_cases}
+# Remove rows that are not complete (e.g., contain NAs) by filtering to only 
+# complete rows
+vs_low_df <- vs_low_df %>%
+  # The dot here means "all"
+  dplyr::filter(complete.cases(.))
 
-```{r read_in_view_neuron, live = TRUE}
-# Read in and take a look at the neuron markers data frame
-neuron_markers_df <- readr::read_tsv(input_file)
-neuron_markers_df
+# Repeat the process for the other DGE results data frame
+vs_unsorted_df <- vs_unsorted_df %>%
+  dplyr::filter(complete.cases(.))
 ```
 
-In our table of marker genes, we use Ensembl gene identifiers. 
-So we will need to convert our marker genes into either gene symbols or Entrez IDs.
+
+In both tables of DGE results, we use Ensembl gene identifiers. 
+So we will need to convert from these identifiers into either gene symbols or Entrez IDs.
 
 ### Gene identifier conversion
```

Comment:
If we move this before the filtering, we can save some steps (identical gene lists) , and also perhaps make the logic a bit more clear about why we need to do the conversion... no steps between getting the database info and making the identifiers match.

Also, you might move the above paragraphs justifying the conversion under this header?